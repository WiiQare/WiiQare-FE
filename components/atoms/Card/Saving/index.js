import { Fragment, useContext, useState } from 'react';
import { HiArrowLongRight } from 'react-icons/hi2';
import { Dialog, Transition } from '@headlessui/react';
import { useFormik } from 'formik';
import { BiTransferAlt } from 'react-icons/bi';
import { TextField } from '@mui/material';
import * as yup from 'yup';
import { useRouter } from 'next/router';
import { DrawContext } from '../../../../pages/_app';
import { useMutation } from 'react-query';
import LoadingButton from '../../Loader/LoadingButton';
import { addPlan } from '../../../../lib/helper';
import { useSession } from 'next-auth/react';
import Toast from '../../Toast';
import Image from 'next/image';

const SavingCard = ({ title, img, month = 12 }) => {
  const { data: session } = useSession();
  const [isOpen, setIsOpen] = useState(false);
  const [isCalculate, setIsCalculate] = useState(false);
  const [plan, setPlan] = useState({ type: '', amount: 0, currency: 'USD' });
  const [pricing, setPricing] = useState([]);
  const [currency, setCurrency] = useState('USD');
  const [state, setState] = useState({ type: 0, message: '' });
  const router = useRouter();
  const { setSaving, saving } = useContext(DrawContext);

  const closeModal = () => {
    setIsOpen(false);
  };

  const openModal = () => {
    setIsOpen(true);
    setSaving({ ...saving, title, month });
  };

  const calculatePrice = (amount) => {
    let prices = [];

    prices.push({ amount: amount / (month * 31), type: 'DAY', currency });
    prices.push({ amount: amount / (month * 4), type: 'WEEK', currency });
    prices.push({ amount: amount / month, type: 'MONTH', currency });

    setPlan(prices[0]);
    setPricing(prices);

    console.log(saving);
  };

  const onSubmit = (values) => {
    if (Object.keys(values).length == 0) return console.log('Pas de données');

    console.log(values);
    if (parseInt(values.amount) == NaN) {
      setState({ type: 2, message: 'Le prix doit être numérique' });
      setTimeout(() => {
        setState({ type: 0, message: '' });
      }, 3000);

      return null;
    }

    setSaving({ ...saving, target: { amount: values.amount, currency } });
    calculatePrice(parseInt(values.amount));

    console.log(pricing);
    setIsCalculate(true);
  };

  const ValidationSchema = yup.object().shape({
    amount: yup
      .number('Le montant doit être numérique')
      .required('Le montant est requis'),
  });

  const formik = useFormik({
    initialValues: {
      amount: '',
    },
    validationSchema: ValidationSchema,
    onSubmit,
  });

  const renderError = (message) => (
    <p className="text-xs text-red-600 font-light flex items-center gap-1 px-1">
      {message}
    </p>
  );

  const newPlan = useMutation(addPlan, {
    onSuccess: (res) => {
      if (res.code) {
        setState({ type: 2, message: res.message ?? res.description });
        setTimeout(() => {
          setState({ type: 0, message: '' });
        }, 3000);
      } else {
        setSaving({ ...saving, idSaving: res.id });
        router.push('/saving/summary');
      }
    },
  });

  const summaryPage = () => {
    console.log(saving);
    newPlan.mutate({
      user: session.user.data.userId,
      type: title == 'Pour moi' ? 'MOI' : title.toUpperCase(),
      amount: saving.target.amount,
      currency: saving.target.currency,
      frequency: plan.type,
      accessToken: session.accessToken,
    });
  };

  const closeToast = () => {
    setState({ type: 0, message: '' });
  };

  return (
    <>
      {state.type > 0 ? (
        state.type == 2 ? (
          <Toast type={'danger'} message={state.message} close={closeToast} />
        ) : state.type == 1 ? (
          <Toast type={'success'} message={state.message} close={closeToast} />
        ) : (
          <></>
        )
      ) : (
        <></>
      )}
      <div
        onClick={() => openModal()}
        className="order-2 md:order-1 bg-white p-5 md:p-10 flex flex-col justify-center items-center rounded-lg shadow-sm hover:shadow-md duration-200 transition-all cursor-pointer gap-4"
      >
        <div className="p-3 rounded-md border border-blue-200 w-fit">
          <Image src={img} alt="Myself" className="w-20" />
        </div>

        <div className="flex flex-col justify-center items-center">
          <h3 className="uppercase text-xs md:text-sm font-bold">{title}</h3>
          <HiArrowLongRight size={25} />
        </div>
      </div>

      <Transition appear show={isOpen} as={Fragment}>
        <Dialog as="div" className="relative z-50" onClose={closeModal}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <div className="fixed inset-0 bg-black bg-opacity-25" />
          </Transition.Child>
          <div className="flex items-center justify-center min-h-full p-4 text-center">
            <div className="fixed inset-0 overflow-y-auto">
              {/* {state.type > 0 ? state.type == 2 ? <Toast type={"danger"} message={state.message} close={closeToast} /> : (state.type == 1 ? <Toast type={"success"} message={state.message} close={closeToast} /> : <></>) : <></>} */}

              <div className="flex items-center justify-center min-h-full p-4 text-center">
                <Transition.Child
                  as={Fragment}
                  enter="ease-out duration-300"
                  enterFrom="opacity-0 scale-95"
                  enterTo="opacity-100 scale-100"
                  leave="ease-in duration-200"
                  leaveFrom="opacity-100 scale-100"
                  leaveTo="opacity-0 scale-95"
                >
                  <Dialog.Panel className="w-full max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-xl space-y-10">
                    <div className="flex flex-col gap-6">
                      <h3 className="mx-auto uppercase font-bold">
                        <span className="text-orange font-medium">
                          Épargne -{' '}
                        </span>
                        {title}
                      </h3>

                      <div className="space-y-10">
                        <div className="flex gap-4 items-center px-5 justify-center">
                          <Image
                            className="inline-block w-16"
                            src={img}
                            alt="Image Description"
                          />
                          <BiTransferAlt size={30} className="text-gray-400" />
                          <Image
                            className="inline-block w-14"
                            src="https://i.goopics.net/nkg0du.png"
                            alt="Image Description"
                          />
                        </div>

                        <form
                          className="flex w-full gap-8 flex-col"
                          id="object"
                          onSubmit={formik.handleSubmit}
                        >
                          <div className="flex flex-col gap-3">
                            <div className="flex justify-between items-center">
                              <h1 className="font-normal text-md text-gray-400">
                                Choisir la devise :{' '}
                              </h1>

                              <label
                                htmlFor="Toggle4"
                                className="inline-flex items-center rounded-full overflow-hidden cursor-pointer bg-white border-2 border-primary text-primary"
                              >
                                <span
                                  onClick={(e) => setCurrency('CDF')}
                                  className={`px-4 py-2  ${currency == 'CDF'
                                    ? 'bg-primary text-white'
                                    : 'bg-transparent text-primary'
                                    } font-semibold select-none text-sm`}
                                >
                                  CDF
                                </span>
                                <span
                                  onClick={(e) => setCurrency('EUR')}
                                  className={`px-4 py-2  ${currency == 'EUR'
                                    ? 'bg-primary text-white'
                                    : 'bg-transparent text-primary'
                                    } font-semibold select-none text-sm`}
                                >
                                  EUR
                                </span>
                                <span
                                  onClick={(e) => setCurrency('USD')}
                                  className={`px-4 py-2  font-semibold ${currency == 'USD'
                                    ? 'bg-primary text-white'
                                    : 'bg-transparent text-primary'
                                    } select-none text-sm`}
                                >
                                  USD
                                </span>
                              </label>
                            </div>
                            <div>
                              <label
                                htmlFor="hs-inline-leading-pricing-select-label"
                                className="block text-sm font-semibold mb-2"
                              >
                                Votre objectif (Pour {month} mois)
                              </label>
                              <div className="relative">
                                <input
                                  type="number"
                                  id="hs-inline-leading-pricing-select-label"
                                  name="amount"
                                  className={`py-3 px-4 ${currency == 'USD' || currency == 'EUR'
                                    ? 'pl-9'
                                    : 'pl-16'
                                    } pr-5 block w-full border-gray-200 shadow-sm rounded-md text-lg focus:z-10 focus:border-blue-500 focus:ring-blue-500 focus:ring-1 outline-none`}
                                  placeholder="0.00"
                                  {...formik.getFieldProps('amount')}
                                />
                                <div className="absolute inset-y-0 left-0 flex items-center pointer-events-none z-20 pl-4">
                                  <span className="text-gray-500 font-semibold text-lg">
                                    {currency == 'USD'
                                      ? '$'
                                      : currency == 'EUR'
                                        ? '€'
                                        : 'CDF'}
                                  </span>
                                </div>
                                <div className="absolute inset-y-0 right-0 flex items-center text-gray-500 pr-px">
                                  <label
                                    htmlFor="hs-inline-leading-select-currency"
                                    className="sr-only"
                                  >
                                    Currency
                                  </label>
                                </div>
                              </div>

                              {formik.errors.amount && formik.touched.amount ? (
                                renderError(formik.errors.amount)
                              ) : (
                                <></>
                              )}
                            </div>
                          </div>

                          <div className="flex flex-row-reverse gap-4">
                            <button
                              form="object"
                              type="submit"
                              className="bg-orange text-sm py-3 px-5 rounded-lg effect-up uppercase font-semibold text-white"
                            >
                              Calculer
                            </button>
                          </div>

                          {isCalculate && pricing.length > 0 ? (
                            <div class={`container mx-auto`}>
                              <p class="text-sm text-center text-gray-500">
                                Choisir un plan
                              </p>

                              <h1 class="mt-1 text-md font-semibold text-center text-gray-800 capitalize lg:text-xl">
                                Prix calculer
                              </h1>

                              <div class="mt-8 space-y-8">
                                <div
                                  onClick={() => {
                                    setPlan(pricing[0]);
                                    setSaving({ ...saving, plan: pricing[0] });
                                  }}
                                  class={`flex items-center justify-between max-w-2xl px-4 py-4 mx-auto border cursor-pointer rounded-xl ${plan.type == 'DAY' ? 'border-blue-500' : ''
                                    }`}
                                >
                                  <div class="flex items-center">
                                    <svg
                                      xmlns="http://www.w3.org/2000/svg"
                                      class={`w-5 h-5 sm:h-9 sm:w-9 ${plan.type == 'DAY'
                                        ? 'text-blue-600'
                                        : 'text-gray-400'
                                        }`}
                                      viewBox="0 0 20 20"
                                      fill="currentColor"
                                    >
                                      <path
                                        fill-rule="evenodd"
                                        d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                                        clip-rule="evenodd"
                                      />
                                    </svg>

                                    <div class="flex flex-col mx-5 space-y-1">
                                      <h2 class="text-lg font-medium text-gray-700">
                                        Journalier
                                      </h2>

                                      <div class="px-2 text-xs text-green-400 bg-gray-100 rounded-full sm:py-1 w-fit">
                                        Save 5%
                                      </div>
                                    </div>
                                  </div>

                                  <h2
                                    class={`text-2xl font-semibold sm:text-3xl ${plan.type == 'DAY'
                                      ? 'text-blue-600'
                                      : 'text-gray-500'
                                      }`}
                                  >
                                    {new Intl.NumberFormat('en-US', {
                                      style: 'currency',
                                      currency: pricing[0].currency,
                                    }).format(pricing[0].amount)}
                                    &nbsp;
                                    <span class="text-base font-medium">
                                      /jour
                                    </span>
                                  </h2>
                                </div>

                                <div
                                  onClick={() => {
                                    setPlan(pricing[1]);
                                    setSaving({ ...saving, plan: pricing[1] });
                                  }}
                                  class={`flex items-center justify-between max-w-2xl px-4 py-4 mx-auto border cursor-pointer rounded-xl ${plan.type == 'WEEK' ? 'border-blue-500' : ''
                                    }`}
                                >
                                  <div class="flex items-center">
                                    <svg
                                      xmlns="http://www.w3.org/2000/svg"
                                      class={`w-5 h-5 sm:h-9 sm:w-9 ${plan.type == 'WEEK'
                                        ? 'text-blue-600'
                                        : 'text-gray-400'
                                        }`}
                                      viewBox="0 0 20 20"
                                      fill="currentColor"
                                    >
                                      <path
                                        fill-rule="evenodd"
                                        d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                                        clip-rule="evenodd"
                                      />
                                    </svg>

                                    <div class="flex flex-col mx-5 space-y-1">
                                      <h2 class="text-lg font-medium text-gray-700">
                                        Hebdomadaire
                                      </h2>

                                      <div class="px-2 text-xs text-green-400 bg-gray-100 rounded-full sm:py-1 w-fit">
                                        Save 20%
                                      </div>
                                    </div>
                                  </div>

                                  <h2
                                    class={`text-2xl font-semibold sm:text-3xl ${plan.type == 'WEEK'
                                      ? 'text-blue-600'
                                      : 'text-gray-500'
                                      }`}
                                  >
                                    {new Intl.NumberFormat('en-US', {
                                      style: 'currency',
                                      currency: pricing[1].currency,
                                    }).format(pricing[1].amount)}
                                    &nbsp;
                                    <span class="text-base font-medium">
                                      /Sem.
                                    </span>
                                  </h2>
                                </div>

                                <div
                                  onClick={() => {
                                    setPlan(pricing[2]);
                                    setSaving({ ...saving, plan: pricing[2] });
                                  }}
                                  class={`flex items-center justify-between max-w-2xl px-4 py-4 mx-auto border cursor-pointer rounded-xl ${plan.type == 'MONTH'
                                    ? 'border-blue-500'
                                    : ''
                                    }`}
                                >
                                  <div class="flex items-center">
                                    <svg
                                      xmlns="http://www.w3.org/2000/svg"
                                      class={`w-5 h-5 sm:h-9 sm:w-9 ${plan.type == 'MONTH'
                                        ? 'text-blue-600'
                                        : 'text-gray-400'
                                        }`}
                                      viewBox="0 0 20 20"
                                      fill="currentColor"
                                    >
                                      <path
                                        fill-rule="evenodd"
                                        d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                                        clip-rule="evenodd"
                                      />
                                    </svg>

                                    <div class="flex flex-col mx-5 space-y-1">
                                      <h2 class="text-lg font-medium text-gray-700">
                                        Mensuel
                                      </h2>

                                      <div class="px-2 text-xs text-green-400 bg-gray-100 rounded-full sm:py-1 w-fit">
                                        Save 33%
                                      </div>
                                    </div>
                                  </div>

                                  <h2
                                    class={`text-2xl font-semibold sm:text-3xl ${plan.type == 'MONTH'
                                      ? 'text-blue-600'
                                      : 'text-gray-500'
                                      }`}
                                  >
                                    {new Intl.NumberFormat('en-US', {
                                      style: 'currency',
                                      currency: pricing[2].currency,
                                    }).format(pricing[2].amount)}
                                    &nbsp;
                                    <span class="text-base font-medium">
                                      /Mois
                                    </span>
                                  </h2>
                                </div>

                                <div class="flex justify-center">
                                  <button
                                    type="button"
                                    className="px-8 py-3 tracking-wide text-white capitalize transition-colors duration-300 transform bg-blue-600 rounded-xl hover:bg-blue-500 focus:outline-none focus:bg-blue-500 focus:ring focus:ring-blue-300 focus:ring-opacity-80"
                                    onClick={() => summaryPage()}
                                  >
                                    Choisir un plan
                                  </button>
                                </div>
                              </div>
                            </div>
                          ) : (
                            <></>
                          )}
                        </form>
                      </div>
                    </div>
                  </Dialog.Panel>
                </Transition.Child>
              </div>
            </div>
          </div>
        </Dialog>
      </Transition>
    </>
  );
};

export default SavingCard;
