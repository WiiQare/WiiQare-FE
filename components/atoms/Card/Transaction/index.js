import { CiCircleList, CiSaveDown2 } from "react-icons/ci";

import DropdownFilter from "../../../atoms/Dropdown/Filter";
import { Button } from "flowbite-react";
const CardTransaction = () => {
  return (
    <div className="border w-full rounded-lg py-8 px-6 flex flex-col flex-wrap md:flex-row justify-between gap-4 bg-white cursor-pointer">
      <div className="">
        <h1 className="md:text-2xl text-gray-700 font-bold flex-wrap flex items-center gap-2 text-lg">
          Transactions History
        </h1>
      </div>
      <div className="flex flex-row flex-wrap  gap-2 center justify-end space-x-2">
        <DropdownFilter
          label="Sort By"
          className="w-[auto]"
          labelClassName="py-1 w-[auto] border-0"
          icon={() => <span className="h-[15px] w-[15px] rounded bg-primary" />}
          items={["Detail", "Cancel"]}
        />
        <Button className="bg-primary">
          <CiSaveDown2 className="mr-2 h-5 w-5" />
          Download Report
        </Button>
        <DropdownFilter
          label="Sort By"
          className="w-[auto]"
          labelClassName="py-1 w-[auto]"
          icon={() => <CiCircleList />}
          items={["Detail", "Cancel"]}
        />
      </div>
    </div>
  );
};

export default CardTransaction;
