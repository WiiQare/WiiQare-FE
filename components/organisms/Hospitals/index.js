import React, { createContext, useState } from 'react';
import CardHeader from '../../atoms/Card/Header';
import { CiCircleList } from 'react-icons/ci';
import DataTable from 'react-data-table-component';
import { useSession } from 'next-auth/react';
import Fetcher from '../../../lib/Fetcher';
import { FcAddressBook, FcCellPhone } from 'react-icons/fc';
import Link from 'next/link';

const businessTypes = {
  MEDICAL_CABINET: { text: 'Cabinet Médical', color: '447698' },
  CLINIC: { text: 'Clinique', color: 'A143D3' },
  HOSPITAL: { text: 'Hôpital', color: 'FF5498' },
};

const columns = [
  {
    name: "IDENTITÉ DE L'HÔPITAL",
    selector: (row) => (
      <div className="flex py-4 gap-4 w-fit">
        <div className="min-w-20 w-20 h-fit border p-3 rounded-lg flex justify-center items-center">
          <img src={row.logoLink} alt="" />
        </div>

        <div className="flex flex-col gap-1">
          <span className="uppercase font-semibold text-md">{row.name}</span>

          <span className="text-xs text-gray-500">{row.email}</span>

          <span className="text-xs text-gray-500">{row.address}</span>

          <span className="flex gap-3 items-center text-sm text-gray-700 font-light mt-3">
            <span className="bg-orange opacity-90 px-2 py-1 text-white text-xs rounded-md w-fit flex gap-3 items-center">
              {row.packages ? row.packages.length : 0} Plan
              {row.packages.length > 1 ? 's' : ''}
            </span>
          </span>
        </div>
      </div>
    ),
  },

  {
    name: 'DÉTAILS BUSINESS',
    selector: (row) => (
      <div className="flex flex-col gap-4">
        <span
          className={`bg-gray-300 text-gray-900 py-1 pl-2 font-semibold w-fit pr-4 rounded-full opacity-80 `}
        >
          {businessTypes[row.businessType].text}
        </span>
        <div className="flex flex-col gap-1">
          <Link href={`tel:${row.phone}`} legacyBehavior>
            <span className="text-xs text-gray-500 flex items-center gap-1 cursor-pointer">
              <FcCellPhone size={18} /> {row.phone}
            </span>
          </Link>

          <Link href={`mailto:${row.email}`} legacyBehavior>
            <span className="text-xs text-gray-500 flex items-center gap-1 cursor-pointer">
              <FcAddressBook size={18} /> {row.email}
            </span>
          </Link>
        </div>
      </div>
    ),
  },

  {
    name: 'ACTIONS',
    button: true,
    cell: (row) => (
      <div>
        <button
          className="border border-gray-800 hover:bg-gray-800 hover:text-gray-200 font-semibold px-3 py-2 text-gray-800 text-xs rounded-md"
          onClick={(e) => alert(row.id)}
        >
          Packages
        </button>
      </div>
    ),
  },
];

const Hospital = () => {
  let { data: session } = useSession();
  const { data, isLoading, isError } = Fetcher(
    `/provider/list`,
    session.user.data.access_token,
  );

  return (
    <div className="p-2 space-y-6 md:py-8 md:px-6 mb-20 md:mb-6">
      <CardHeader
        title={'Liste des hôpitaux'}
        filter={{
          label: {
            title: 'Trier par',
            className: 'py-1 w-[auto]',
          },
          className: 'w-[auto]',
          icon: () => <CiCircleList />,
          items: ['Detail', 'Annuler'],
        }}
      />

      <div className="border rounded-lg overflow-x-auto w-full relative">
        {isLoading ? (
          <>Loading...</>
        ) : (
          <>
            {data.length > 0 ? (
              <DataTable
                columns={columns}
                data={data}
                fixedHeader
                className="px-8 bg-white"
                title="Liste des hôpitaux"
                persistTableHead
              />
            ) : (
              <div className="py-10 mx-auto w-1/2 flex flex-col justify-center items-center gap-4">
                <p className="text-gray-500 text-sm">
                  Aucun hôpital actuellement ajouté...
                </p>
              </div>
            )}
          </>
        )}
      </div>
    </div>
  );
};

export default Hospital;
