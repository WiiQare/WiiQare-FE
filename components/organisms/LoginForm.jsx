import React from "react";
import Link from "next/link";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepButton from "@mui/material/StepButton";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import useMediaQuery from "@mui/material/useMediaQuery";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";

import OnboardingScreen from "../molecules/OnboardingScreen";
import { useLocalStorage } from "../../hooks/useLocalStorage";
import {
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  Typography,
} from "@mui/material";
import MenuHolder from "../atoms/MenuHolder";
import BlinkSnackbar from "../atoms/BlinkSnackbar";

function LoginForm() {
  const matches = useMediaQuery("(max-width: 992px)");
  const [onboardingIsEnd, setOnboardingIsEnd] = React.useState("");
  const [showPassword, setShowPassword] = React.useState(false);

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  React.useEffect(() => {
    setOnboardingIsEnd(localStorage.getItem("onboardingIsEnd") || "");
  }, []);

  const formHolder = (
    <>
      <div className="form-holder">
        <MenuHolder href="/register" label="SIGN UP" />
        <div className="signin-signup-form">
          <div className="form-items">
            <div className="form-title">Sign in to your account</div>
            <Box sx={{ mb: 3, mt: 2 }}></Box>
            <form id="signinform">
              <Stack spacing={2}>
                <TextField
                  id="outlined-basic"
                  fullWidth
                  label="E-mail Address"
                  variant="outlined"
                />
                <FormControl fullWidth variant="outlined">
                  <InputLabel htmlFor="outlined-basic1">Password</InputLabel>
                  <OutlinedInput
                    id="outlined-basic1"
                    label="Password"
                    type={showPassword ? "text" : "password"}
                    endAdornment={
                      <InputAdornment position="end">
                        <IconButton
                          aria-label="toggle password visibility"
                          onClick={handleClickShowPassword}
                          edge="end"
                        >
                          {showPassword ? <VisibilityOff /> : <Visibility />}
                        </IconButton>
                      </InputAdornment>
                    }
                  />
                </FormControl>
                <Box sx={{ textAlign: "right", mb: 2 }}>
                  <Typography color="text.secondary" variant="caption">
                    Forgot your password ?
                  </Typography>
                </Box>
                <Box>
                  <Button
                    size="large"
                    variant="contained"
                    onClick={() => {}}
                    className="bg-yellow"
                  >
                    SIGN IN
                  </Button>
                </Box>
              </Stack>
            </form>
          </div>
        </div>
      </div>
      <BlinkSnackbar />
    </>
  );
  return (
    <>
      {matches && !onboardingIsEnd ? (
        <OnboardingScreen
          onStartClick={() => {
            if (typeof window !== "undefined") {
              localStorage.setItem("onboardingIsEnd", "onboardingIsEnd");
            }
            setOnboardingIsEnd("onboardingIsEnd");
          }}
        />
      ) : (
        formHolder
      )}
    </>
  );
}

export default LoginForm;
