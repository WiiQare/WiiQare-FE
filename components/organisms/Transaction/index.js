import React, { createContext, useState } from "react";
import CardHeader from "../../atoms/Card/Header";
import HistoryWallet from "../Wallet/History";
import { CiCircleList } from "react-icons/ci";

import Quickly from "./quickly";
export const TransactionContext = createContext();


const Transaction = () => {

	const [transaction, setTransaction] = useState({state: false, transactions: []});

	return (
		<div className="p-2 space-y-6 md:py-8 md:px-6 mb-20">
			<CardHeader
				title={"Historique des transactions"}
				sort={{
					icon: () => <span className="h-[15px] w-[15px] rounded bg-primary" />,
					items: ["DÃ©tail", "Annuler"]
				}}
				filter={{
					label: {
						title: "Trier par",
						className: "py-1 w-[auto]"
					},
					className: "w-[auto]",
					icon: () => <CiCircleList />,
					items: ["Detail", "Annuler"]
				}}
			/>

			<TransactionContext.Provider value={{ transaction, setTransaction }}>
				<HistoryWallet />
				<Quickly />
			</TransactionContext.Provider>

		</div>
	);
};

export default Transaction;
