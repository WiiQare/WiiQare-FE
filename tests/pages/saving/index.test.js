import Saving from '@/pages/saving';
import { render } from '@testing-library/react';
import { TransactionContext } from '@/components/organisms/Transaction';
import { SessionProvider } from 'next-auth/react';

describe('Saving', () => {
  it('should render', () => {
    const { container } = render(
      <SessionProvider session={{ user: { data: { access_token: {} } } }}>
        <TransactionContext.Provider
          value={{ transaction: {}, setTransaction: () => {} }}
        >
          <Saving />
        </TransactionContext.Provider>
      </SessionProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
