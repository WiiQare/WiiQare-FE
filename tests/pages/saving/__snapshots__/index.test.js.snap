// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saving should render 1`] = `
<div>
  <div
    class="p-2 space-y-6 md:py-8 md:px-6 mb-12"
  >
    <div
      class="border w-full rounded-lg py-8 px-6 flex flex-col flex-wrap md:flex-row justify-between items-center gap-4 bg-white cursor-pointer"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="md:text-2xl text-gray-700 font-bold flex-wrap flex items-center gap-2 text-lg"
        >
          Épargne de Santé
        </h1>
      </div>
      <div
        class="flex flex-row flex-wrap  gap-2 center justify-end space-x-2"
      >
        <button
          class="text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg bg-primary"
          type="button"
        >
          <span
            class="flex items-center rounded-md text-sm px-4 py-2"
          >
            <svg
              class="mr-2 h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Square_Plus"
              >
                <g>
                  <path
                    d="M18.438,20.938H5.563a2.5,2.5,0,0,1-2.5-2.5V5.564a2.5,2.5,0,0,1,2.5-2.5H18.438a2.5,2.5,0,0,1,2.5,2.5V18.438A2.5,2.5,0,0,1,18.438,20.938ZM5.563,4.064a1.5,1.5,0,0,0-1.5,1.5V18.438a1.5,1.5,0,0,0,1.5,1.5H18.438a1.5,1.5,0,0,0,1.5-1.5V5.564a1.5,1.5,0,0,0-1.5-1.5Z"
                  />
                  <path
                    d="M15,12.5H12.5V15a.5.5,0,0,1-1,0V12.5H9a.5.5,0,0,1,0-1h2.5V9a.5.5,0,0,1,1,0v2.5H15A.5.5,0,0,1,15,12.5Z"
                  />
                </g>
              </g>
            </svg>
            Nouvelle épargne
          </span>
        </button>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="bg-white drop-shadow-sm rounded-lg px-3 py-3"
      >
        <div
          class="flex flex-row-reverse"
        >
          <div
            class="undefined dropdown dropdown-end flex"
          >
            <label
              class=" bg-white border rounded-xl w-full py-2 px-4 flex justify-between items-center border-none"
              tabindex="0"
            >
              <span
                class="flex mr-2 gap-4 items-center"
              >
                <svg
                  aria-hidden="true"
                  class="cursor-pointer text-gray-700"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                  />
                </svg>
                <span
                  class="font-light"
                />
              </span>
            </label>
            <ul
              class="!p-0 mt-10 w-28 w-full mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light"
              tabindex="0"
            >
              <li>
                <a>
                  Delete
                </a>
              </li>
              <li>
                <a>
                  Edit
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="flex flex-col md:flex-row gap-6 md:items-center justify-between px-6"
        >
          <div
            class="flex gap-2 items-center"
          >
            <svg
              class="text-sky"
              fill="currentColor"
              height="80"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Wallet"
              >
                <g>
                  <path
                    d="M19.435,4.065H4.565a2.5,2.5,0,0,0-2.5,2.5v10.87a2.5,2.5,0,0,0,2.5,2.5h14.87a2.5,2.5,0,0,0,2.5-2.5V6.565A2.5,2.5,0,0,0,19.435,4.065Zm1.5,9.93h-6.42a2,2,0,0,1,0-4h6.42Zm-6.42-5a3,3,0,0,0,0,6h6.42v2.44a1.5,1.5,0,0,1-1.5,1.5H4.565a1.5,1.5,0,0,1-1.5-1.5V6.565a1.5,1.5,0,0,1,1.5-1.5h14.87a1.5,1.5,0,0,1,1.5,1.5v2.43Z"
                  />
                  <circle
                    cx="14.519"
                    cy="11.996"
                    r="1"
                  />
                </g>
              </g>
            </svg>
            <div
              class="text-gray-700"
            >
              <h1
                class="font-semibold text-xl"
              >
                Portefeuille ID
              </h1>
              <h3
                class="font-light uppercase text-md"
              >
                #
                ...
              </h3>
            </div>
          </div>
          <span
            class="w-12 hidden md:block"
          />
        </div>
        <div
          class="px-6 mt-8 space-y-7"
        >
          <div
            class="flex flex-col justify-center items-center my-10 md:my-20 gap-6"
          >
            <img
              alt=""
              class="md:w-44 w-28 opacity-80"
              data-nimg="1"
              decoding="async"
              height="512"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fi.goopics.net%2Fozsga7.png&w=1080&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.goopics.net%2Fozsga7.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fi.goopics.net%2Fozsga7.png&w=1080&q=75 2x"
              style="color: transparent;"
              width="512"
            />
            <span
              class="text-gray-400 text-sm"
            >
              Aucun épargne actuellement en cours...
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
